ambry-api/src/main/java/com.github.ambry/account/Account.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * A representation of an Ambry user. A user is an entity (an application or an individual user) who uses ambry
+ * as service. This class contains general information of a user, which can be used for user-based operations
+ * such as authentication, get a {@link Container} under this account, and access control. The account name is
+ * provided by an Ambry user as an external reference. Account id is an internal identifier of the user, and is
+ * one-to-one mapped to the account name. Account name and id are generated through user registration process.
+ * Account id is part of a blob’s properties, and cannot be modified once the blob is created. Account metadata
+ * is made in JSON, which is generic to contain additional information of the metadata.
+ *
+ * Version 1 of account metadata in JSON is in the format below.
+ *  {
+ *    "accountId": 101,
+ *    "accountName": "MyAccount",
+ *    "containers": [
+ *      {
+ *        "containerName": "MyPrivateContainer",
+ *        "description": "This is my private container",
+ *        "isPublic": "true",
+ *        "containerId": 0,
+ *        "version": 1,
+ *        "status": "active"
+ *      },
+ *      {
+ *        "containerName": "MyPublicContainer",
+ *        "description": "This is my public container",
+ *        "acl": "false",
+ *        "containerId": 1,
+ *        "version": 1,
+ *        "status": "active"
+ *      }
+ *    ],
+ *    "version": 1,
+ *    "status": "active"
+ *  }
+ */
+public class Account {
```

Comment:
In the initial design, both `Account` and `Container` are abstracted as interface. The reason to concretize it is they are data structure rather than function components. Versioning and the adoption of json metadata provide a flexible way to extend to a different structure for the two classes. In your example, the different structure of `Container` can be made by changing  the metadata of the `Container` only, and inserting the metadata into the metadata of `Account`. Constructing and parsing `Container` metadata is handled by `Container` itself.